java/src/jmri/jmrix/mqtt/MqttReporter.java

```diff
@@ -0,0 +1,67 @@
+package jmri.jmrix.mqtt;
+
+import java.util.HashSet;
+
+import jmri.IdTagManager;
+import jmri.InstanceManager;
+import jmri.IdTag;
+
+import jmri.implementation.AbstractIdTagReporter;
+
+/**
+ * Provide a Reporter implementation for MQTT communications
+ *
+ * @author Bob Jacobsen Copyright (C) 2023
+ */
+
+class MqttReporter extends AbstractIdTagReporter implements MqttEventListener {
+
+    /**
+     * Requires, but does not check, that the system name and topic be consistent
+     * @param ma Adapter to specific connection
+     * @param systemName System Name for this Sensor
+     * @param rcvTopic Topic string to be used when receiving by JMRI
+     */
+    MqttReporter(MqttAdapter ma, String systemName, String rcvTopic) {
+        super(systemName);
+        this.rcvTopic = rcvTopic;
+        mqttAdapter = ma;
+        mqttAdapter.subscribe(rcvTopic, this);
+    }
+
+    private final MqttAdapter mqttAdapter;
+    private final String rcvTopic;
+
+    @Override
+    public void notifyMqttMessage(String receivedTopic, String message) {
+        if (! receivedTopic.endsWith(rcvTopic) ) {
+            log.error("Got a message whose topic ({}) wasn't for me ({})", receivedTopic, rcvTopic);
+            return;
+        }
+
+        log.trace("start parse of {}", message);
+
+        // parse content
+        String[] terms = message.split(" ", 2);
+
+        if ( terms.length < 1 || terms[0].isEmpty()) {
+            log.debug("No loco ID present in ({}), record empty report", message);
+            notify(null);
+            return;
+        }
+        // normal condition
+        String loco = terms[0];
+
+        String content  = "";
+        if (terms.length > 1 ) {
+            content = terms[1];
+        }
+
+        IdTag idTag = InstanceManager.getDefault(IdTagManager.class).provideIdTag(loco);
+        idTag.setProperty("content", content);
+        notify(idTag);
```

Comment:
What will happen if IdTagManager.provideIdTag() throws an IllegalArgumentException? It seems to me that that would happen if the loco has the ID that's equal to the system connection name, for example if the system connection is M2 and the loco in this message has the id M2.

In that case, this method will throw an IllegalArgumentException. Will that cause any problem for the user? Just checking.

Reply:
Good point. Will catch and warn.